ctera/src/test/java/ch/cyberduck/core/ctera/CteraMoveFeatureTest.java

```diff
@@ -0,0 +1,175 @@
+        assertFalse(new DAVFindFeature(session).find(test));
+        assertTrue(new DAVFindFeature(session).find(target));
+        assertEquals(status.getResponse(), target.attributes());
+        assertEquals(attr, new CteraAttributesFinderFeature(session).find(target));
+        new CteraDeleteFeature(session).delete(Collections.<Path>singletonList(target), new DisabledLoginCallback(), new Delete.DisabledCallback());
+    }
+
+    @Test
+    public void testMoveDirectory() throws Exception {
+        final Path folder = new Path(new DefaultHomeFinderService(session).find(), new AlphanumericRandomStringService().random(), EnumSet.of(Path.Type.directory));
+        new CteraDirectoryFeature(session).mkdir(folder, new TransferStatus());
+        final Path test = new Path(folder, new AlphanumericRandomStringService().random(), EnumSet.of(Path.Type.file));
+        {
+            final byte[] content = RandomUtils.nextBytes(3547);
+            final TransferStatus status = new TransferStatus();
+            status.setOffset(0L);
+            status.setLength(1024L);
+            final HttpResponseOutputStream<Void> out = new CteraWriteFeature(session).write(test, status, new DisabledConnectionCallback());
+            // Write first 1024
+            new StreamCopier(status, status).withOffset(status.getOffset()).withLimit(status.getLength()).transfer(new ByteArrayInputStream(content), out);
+            out.close();
+        }
+        final PathAttributes attr = new CteraAttributesFinderFeature(session).find(test);
+        final Path target = new Path(new DefaultHomeFinderService(session).find(), new AlphanumericRandomStringService().random(), EnumSet.of(Path.Type.directory));
+        new CteraMoveFeature(session).move(folder, target, new TransferStatus(), new Delete.DisabledCallback(), new DisabledConnectionCallback());
+        assertFalse(new DAVFindFeature(session).find(folder));
+        assertFalse(new DAVFindFeature(session).find(test));
+        assertTrue(new DAVFindFeature(session).find(target));
+        assertTrue(new DAVFindFeature(session).find(new Path(target, test.getName(), EnumSet.of(Path.Type.file))));
+        assertEquals(attr, new CteraAttributesFinderFeature(session).find(new Path(target, test.getName(), EnumSet.of(Path.Type.file))));
+        assertEquals(attr.getModificationDate(), new CteraAttributesFinderFeature(session).find(new Path(target, test.getName(), EnumSet.of(Path.Type.file))).getModificationDate());
+        assertEquals(attr.getFileId(), new CteraAttributesFinderFeature(session).find(new Path(target, test.getName(), EnumSet.of(Path.Type.file))).getFileId());
+        // TODO CTERA-137 what is the expected behaviour?
+        //assertNotEquals(attr.getETag(), new CteraAttributesFinderFeature(session).find(new Path(target, test.getName(), EnumSet.of(Path.Type.file))).getETag());
```

Comment:
@dkocher what is the expected behaviour?

Reply:
ETag should remain constant when moving a resource but Ctera may not be WebDAV compliant.

Reply:
4528a68f1e01f53f7db13f3927a9dc6e085cc877